hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/table/functional/TestHoodieSparkMergeOnReadTableCompaction.java

```diff
@@ -200,11 +221,190 @@ public void testWriteLogDuringCompaction(boolean enableMetadataTable, boolean en
       client.delete(deleteRecords, client.startCommit());
       // insert the same 100 records again
       client.upsert(writeRecords, client.startCommit());
-      Assertions.assertEquals(100, readTableTotalRecordsNum());
+      assertEquals(100, readTableTotalRecordsNum());
     } finally {
       jsc().hadoopConfiguration().set(HoodieReaderConfig.FILE_GROUP_READER_ENABLED.key(), "true");
     }
+  }
+
+  @ParameterizedTest
+  @CsvSource(value = {"true,true", "true,false", "false,true", "false,false"})
+  void testCompactionSchedulingWithUncommittedLogFileOrRollback(boolean enableMetadataTable,
```

Comment:
Could you explain the particular logic this test covers?  And is it possible to add a unit test instead of the functional test here?  Same on reporting the test time.

Reply:
It was added as part of HUDI-8123 which makes sure that marker based rollback does not include non existent log files. This PR cherry-picks that commit as well.

Reply:
I added the javadocs for the test.